### PR TITLE
Fix local tests

### PR DESCRIPTION
**Pull request by eevelweezel**
## What do these changes do?

Fix tests to work locally with later versions of Docker.

## Are there changes in behavior for the user?

No

## Related issue number
#849, possibly #691

## Checklist

- [ x] I think the code is well written
- [ x] Unit tests for the changes exist
- [ x] Documentation reflects the changes
- [ x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [ x] Add a new news fragment into the `changes` folder
  * name it `<issue_id>.<type>` for example (588.bug)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by achimnol** [comment]
FYI: https://github.com/actions/actions-runner-controller/blob/master/runner/actions-runner.ubuntu-22.04.dockerfile#L8
The Docker version in the action runner `ubuntu-latest` is 24.0.7 currently.

**Comment by achimnol** [comment]
In my setup, there are more remaining failure cases:
```
FAILED tests/test_containers.py::test_commit - KeyError: 'Container'
FAILED tests/test_containers.py::test_commit_with_changes - KeyError: 'Container'
FAILED tests/test_images.py::test_build_from_remote_tar - aiodocker.exceptions.DockerError: DockerError(404, 'No such image: aiodocker-75555cf:1.0')
FAILED tests/test_networks.py::test_list_networks_with_filter - aiodocker.exceptions.DockerError: DockerError(409, 'network with name test-net-filter already exists')
FAILED tests/test_volumes.py::test_create_show_delete_volume[False] - IndexError: list index out of range

Results (106.40s (0:01:46)):
      98 passed
       4 failed
         - tests/test_containers.py:168 test_commit
         - tests/test_containers.py:186 test_commit_with_changes
         - tests/test_images.py:40 test_build_from_remote_tar
         - tests/test_networks.py:14 test_list_networks_with_filter
       1 error
delete:
aiodocker-test-registry
delete:
aiodocker-test-registry2
```

File: tests/conftest.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ async def random_name():
     docker = Docker()
     images = await docker.images.list()
     for img in images:
-        if img["RepoTags"] is None:
+        if img["RepoTags"] in [None, []]:
```
Comment (achimnol) [inline tests/conftest.py:55]: This could be simplified as:
```suggestion
        if img["RepoTags"]:
```

Comment (achimnol) [inline tests/conftest.py:55]: I'd suggest simplifying this as:
```suggestion
        if img["RepoTags"]:
```

File: tests/test_networks.py [hunk 2]
```diff
@@ -13,12 +13,17 @@ async def test_list_networks(docker):
 
 @pytest.mark.asyncio
 async def test_list_networks_with_filter(docker):
-    await docker.networks.create({
+    network = await docker.networks.create({
         "Name": "test-net-filter",
-        "Labels": {"some": "label"},
+        "labels": {"some": "label"},
```
Comment (achimnol) [inline tests/test_networks.py:18]: According to the docker API reference, this field starts with an uppercase letter, though Docker seems to accept both lowercase and uppercase letters: https://docs.docker.com/engine/api/v1.45/#tag/Network/operation/NetworkCreate

**Review by achimnol — CHANGES_REQUESTED** [review]
I'm going to merge the PR and inspect other local testing issues in my setup as a follow-up. Please check out my review comments!

File: tests/test_networks.py (hunk 2 above)
Comment (eevelweezel) [reply tests/test_networks.py:18]: This was an accident, I'll revert.

File: tests/conftest.py (hunk 1 above)
Comment (eevelweezel) [reply tests/conftest.py]: We want the false-y case, so I changed this to `if not img["RepoTags"]:`.

Comment (achimnol) [reply tests/conftest.py]: Oh, that's correct. Thanks!

**Review by achimnol — APPROVED** [review]

**Pull request merged**